File: app/src/main/java/net/theluckycoder/scriptcraft/utils/CustomTabWidthSpan.java
package net.theluckycoder.scriptcraft.utils;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.annotation.NonNull;
import android.text.style.ReplacementSpan;

public final class CustomTabWidthSpan extends ReplacementSpan {

    public CustomTabWidthSpan(){}


    @Override
    public int getSize(@NonNull Paint paint, CharSequence text, int start, int end, Paint.FontMetricsInt fm) {
        return 20;
    }

    @Override
    public void draw(@NonNull Canvas canvas, CharSequence text, int start, int end, float x, int top, int y, int bottom, @NonNull Paint paint) {

    }

}

File: app/src/main/java/net/theluckycoder/sharpide/utils/CustomTabWidthSpan.kt
package net.theluckycoder.sharpide.utils

import android.graphics.Canvas
import android.graphics.Paint
import android.text.style.ReplacementSpan

class CustomTabWidthSpan : ReplacementSpan() {

    override fun getSize(paint: Paint, text: CharSequence, start: Int, end: Int, fm: Paint.FontMetricsInt?): Int = 20

    override fun draw(canvas: Canvas, text: CharSequence, start: Int, end: Int, x: Float, top: Int, y: Int, bottom: Int, paint: Paint) { }

}
